src/mkdocstrings/handlers/base.py

```diff
@@ -184,6 +187,8 @@ def do_heading(
         if toc_label is None:
             toc_label = content.unescape() if isinstance(el, Markup) else content  # type: ignore
         el.set("data-toc-label", toc_label)
+        if role:
+            el.set("data-role", role)
```

Comment:
Note that the "data-role" never ends up in any HTML, and this information could've been associated with the item in any other way. But yeah this is the most straightforward way that we have now.

Reply:
With my basic knowledge of HTML, I thought you could put anything in `data-*` attributes :slightly_smiling_face: 
I'm good with it just being a temporary data holder.